nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/PutS3Object.java

```diff
@@ -102,6 +177,94 @@ protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String
             .build();
     }
 
+    protected File getPersistenceFile() {
+        return new File(PERSISTENCE_ROOT + getIdentifier());
+    }
+
+    @Override
+    public void onPropertyModified(final PropertyDescriptor descriptor, final String oldValue, final String newValue) {
+        if (descriptor.equals(KEY)
+                || descriptor.equals(BUCKET)
+                || descriptor.equals(ENDPOINT_OVERRIDE)
+                || descriptor.equals(STORAGE_CLASS)
+                || descriptor.equals(REGION)) {
+            destroyState();
+        }
+    }
+
+    protected MultipartState getState(final String s3ObjectKey) throws IOException {

```

Comment:
I believe these methods could be problematic if multiple threads are monkeying with the persistence file at the same time.